### ISSUE TITLE
All Play Benchmarks Fail: bind address fix

### ISSUE THREAD

**Issue by godenji** [TechEmpower/FrameworkBenchmarks#1553]
See if adding `-Dhttp.address=...` to the respective start script[1]s fixes the issue (Play binds to `0.0.0.0` and port 9000 by default). Failing that, maybe `-Dhttps.port` needs to be set if the benchmarks are run over https.

Here's Play error handling section corresponding to TE log entry [2]

[1] https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/Scala/play2-scala/setup_scala.sh#L18

[2] https://github.com/playframework/playframework/blob/2.2.x/framework/src/play/src/main/scala/play/core/server/NettyServer.scala#L295

**Comment by Rydgel** [comment]
Noticed that too. A rematch would be nice.

**Comment by eranation** [comment]
:+1: for a rematch

**Comment by bhauer** [comment]
I'd like others to take a closer look, but I believe this is a toolset problem (more specifically, a misbehaving framework unrelated to Play creating processes under another username and then not cleaning up and also avoiding our aggressive toolset clean-up).  I suspect it's not a Play issue.

**Comment by richdougherty** [comment]
My vote is to fix the underlying problem of tests not being cleaned up instead of changing Play to bind to a different port. I think it's OK for Play to fail if other tests are still running. We don't want to benchmark Play if other tests are still running because the other tests might use CPU or memory resources and affect the Play result. I think a bind failure is a good outcome if other tests are still running.

As well as cleaning up after tests properly, it would also be helpful if we could see Play's stderr output so we can verify that it is a bind error causing the problem.

**Comment by richdougherty** [comment]
Just wondering if there is any progress on finding the cause of this error. Is there another ticket I can follow to keep an eye on progress?

In https://github.com/TechEmpower/FrameworkBenchmarks/issues/1545#issuecomment-99019344 there was a suggestion of dumping _netstat_ to the logs to help track down the misbehaving test(s). That sounds like a good idea ot me.

**Comment by msmith-techempower** [comment]
This issue is fixed, but we have run into other blocking issues with round 11; we will have more details soon.

**Comment by hamiltont** [comment]
Can we point to the commit where this was fixed and close this? Or perhaps it's not in master yet?

**Comment by richdougherty** [comment]
I'd be interested to hear more about how this was fixed. Was there a change to how processes are terminated at the end of each test?

**Comment by msmith-techempower** [comment]
Er.. this was fixed in #1605 

The problem was the following calamity:
1. We get a fresh OS install every time we run a round
2. We ran the full suite and `hhvm` was installed via `apt-get` (booo! hisss!)
3. `apt-get install` decided that "install" means "install and turn on with default config file", so `hhvm` was started and bound to port 9000, but no app code was run, so it's under the wrong user.

Basically, the fix is this https://github.com/TechEmpower/FrameworkBenchmarks/commit/1273e4035c0f9003167f8444373f60c48e7f2a65

**Comment by richdougherty** [comment]
Thanks for explaining. :) Glad to hear this is fixed!

**Comment by richdougherty** [comment]
![Anger](http://gifrific.com/wp-content/uploads/2013/12/Monitor-Throw-Maurice-Moss-The-IT-Crowd.gif)?

**Comment by godenji** [comment]
and still no Play framework in TE 11?? Play 1 is there but that's the legacy version of the framework. What happened to Play 2?

**Comment by msmith-techempower** [comment]
Play2 still had a bug that was resolved after we captured Round 11.

**Comment by wsargent** [comment]
@msmith-techempower What was the bug?  Is there a reference to it?
